vscode-trace-webviews/src/trace-explorer/time-range/vscode-time-range-data-widget.tsx

```diff
@@ -0,0 +1,95 @@
+import 'traceviewer-react-components/style/trace-explorer.css';
+// import { Experiment } from 'tsp-typescript-client/lib/models/experiment';
+import '../../style/react-contextify.css';
+import '../../style/trace-viewer.css';
+import JSONBigConfig from 'json-bigint';
+import { convertSignalExperiment } from 'vscode-trace-common/lib/signals/vscode-signal-converter';
+import { VSCODE_MESSAGES, VsCodeMessageManager } from 'vscode-trace-common/lib/messages/vscode-message-manager';
+import { TimeRangeUpdatePayload } from 'traceviewer-base/lib/signals/time-range-data-signal-payloads';
+
+const JSONBig = JSONBigConfig({
+    useNativeBigInt: true,
+});
+
+interface TimeRangeDataWidgetProps {
+    data: boolean;
+}
+
+class TimeRangeDataWidget extends React.Component {
+  private _signalHandler: VsCodeMessageManager;
+
+  static ID = 'trace-explorer-time-range-data-widget';
+  static LABEL = 'Time Range Data';
+
+  constructor(props: TimeRangeDataWidgetProps) {
+      super(props);
+      this._signalHandler = new VsCodeMessageManager();
+      window.addEventListener('message', event => {
+
+          const { command, data } = event.data;
+
+          const experiment = data.wrapper ? convertSignalExperiment(JSONBig.parse(data.wrapper)) : undefined;
+          const parsedData: TimeRangeUpdatePayload = !experiment ? JSONBig.parse(data) : undefined;
+
+          switch (command) {
+          case VSCODE_MESSAGES.EXPERIMENT_SELECTED:
+              signalManager().fireExperimentSelectedSignal(experiment);
+              break;
+          case VSCODE_MESSAGES.EXPERIMENT_UPDATED:
+              if (experiment) {
+                  signalManager().fireExperimentUpdatedSignal(experiment);
+              }
+              break;
+          case VSCODE_MESSAGES.EXPERIMENT_CLOSED:
+              // TODO - make sure this signal is getting passed.
+              if (experiment) {
+                  signalManager().fireExperimentClosedSignal(experiment);
+              }
+              break;
+          case VSCODE_MESSAGES.SELECTION_RANGE_UPDATED:
+              if (experiment) {
+                  signalManager().fireSelectionRangeUpdated(parsedData);
+              }
+              break;
+          case VSCODE_MESSAGES.VIEW_RANGE_UPDATED:
+              if (experiment) {
+                  signalManager().fireViewRangeUpdated(parsedData);
+              }
+              break;
+          }
+
+      });
+  }
+
+  UNSAFE_componentWillMount(): void {
+      signalManager().on(Signals.REQUEST_SELECTION_RANGE_CHANGE, this.onRequestSelectionChange);
+  }
+
+  componentWillUnmount(): void {
+      signalManager().off(Signals.REQUEST_SELECTION_RANGE_CHANGE, this.onRequestSelectionChange);
+  }
+
+  onRequestSelectionChange = (payload: TimeRangeUpdatePayload): void => {
+      this._signalHandler.requestSelectionRangeChange(payload);
+  };
+
+  componentDidMount(): void {
+      this._signalHandler.notifyReady();
```

Comment:
Is this notification handled?

Reply:
I am not sure what this signal is or where it is handled.  I saw that every other webview that is in the trace-explorer fired this signal so I left it in mine.  @bhufmann, offhand, do you know if this is necessary or not?

Reply:
Usually a signal like this would be used to facilitate the following sequence:
 - User opens view
 - View provider creates webview
 - When webview is ready, view provider sends it data to render

If there's no lifecycle handling that requires the plugin side to interact with the webview when it's ready (but not before), then the ready signal is redundant (but doesn't do any harm).

Reply:
I understand that state restoration when the webview goes out of view is a problem for this PR. That is a potential use for this signal, as well. If the provider knows what signals to send to get the view in the right state, it can send those whenever it receives the ready signal. That way, every time the view re-mounts, the provider sends what's necessary to show the correct data in the view.

Alternatively, that could be handled on the view side using the VSCode's webview state restoration mechanisms. In that case, the webview side would save its state as the most recent, state-relevant signals it had received, and then fetch and reemit those signals on remount.